numalogic/preprocess/transformer.py

```diff
@@ -47,3 +50,40 @@ def transform(self, X):
 
     def inverse_transform(self, X) -> ArrayLike:
         return np.power(X, 1.0 / self.n) - self.add_factor
+
+
+class TanhScaler(BaseEstimator):
+    r"""
+    Tanh Estimator applies tanh normalization to the Z-score,
+    and scales the values between 0 and 1.
+
+    After scaling, the data has a mean of 0.5.
+
+    The coeff parameter determines the spread of the scores.
+    Higher the value, the linear portion of the curve will have a higher slope
+    but will reach the asymptote (flatten out) earlier.
+
+    References:
+        Nandakumar, Jain, Ross. 2005. Score Normalization in
+        Multimodal Biometric Systems, Pattern Recognition 38, 2270-2285.
+        https://web.cse.msu.edu/~rossarun/pubs/RossScoreNormalization_PR05.pdf
+    """
+    __slots__ = ("_coeff", "_std", "_mean")
+
+    def __init__(self, coeff: float = 0.2):
+        self._coeff = coeff
+        self._std = None
+        self._mean = None
+
+    def fit(self, x: npt.NDArray[float]) -> Self:
+        self._mean = np.mean(x, axis=0)
+        self._std = np.std(x, axis=0)
+        return self
+
+    def transform(self, x: npt.NDArray[float]) -> npt.NDArray[float]:
+        x_std_scaled = (x - self._mean) / self._std
+        return 0.5 * (np.tanh(self._coeff * x_std_scaled) + 1)
```

Comment:
I think we should still have the parameters passed as an option. We can keep these values as default values

Reply:
You mean the `0.5`?

Reply:
I think it is better to be a little restrictive, since the user might not have much idea about the different parameters. We can always add the flexibility later on.

Reply:
yeah 
0.5 = mean of the new distribution.
0.01 = std of the new distribution.

Someday, we might need to change the mean or spread